TypeCobol.LanguageServer/Workspace.cs

```diff
@@ -260,12 +250,24 @@ private FileCompiler OpenTextDocument(DocumentContext docContext, string sourceT
             return fileCompiler;
         }
 
-        public bool TryGetOpenedDocumentContext(Uri fileUri, out DocumentContext openedDocumentContext)
+        /// <summary>
+        /// Open a Text Document.
+        /// </summary>
+        /// <param name="docContext">The Document Context</param>
+        /// <param name="projectKey">Project's Key</param>
+        /// <param name="sourceText">The source text</param>
+        /// <param name="lsrOptions">LSR testing options</param>
+        /// <returns></returns>
+        private FileCompiler OpenTextDocument(DocumentContext docContext, string projectKey, 
+            string sourceText,
+            List<string> copyFolders, LsrTestingOptions lsrOptions)
         {
-            lock (_lockForOpenedDocuments)
-            {
-                return _openedDocuments.TryGetValue(fileUri, out openedDocumentContext);
-            }
+            var workspaceProject = docContext.Project ?? WorkspaceProjectStore.GetOrCreateProject(projectKey);
```

Comment:
Why checking the Project here ? A non-null project means that the document was already opened and it also means that the projectKey parameter is ignored. I think we should make more apparent the cases where a document is re-opened and where it is opened for the first time.